### PR TITLE
Add float support to sleep()

### PR DESCRIPTION
**Pull request by saveriomiroddi**
Add float support, and also improve the error handling.

### CODE REVIEW

**Comment by saveriomiroddi** [comment]
@st0012 I couldn't find any existing UT for this method, so I haven't added any (I'm not sure if there is a reason for it).

**Comment by saveriomiroddi** [comment]
@st0012 what's your take about this? If merged, I would rebase #493 on top of it, so I can make the sleep intervals much smaller.

**Comment by st0012** [comment]
@saveriomiroddi I'd merge this once the coverage issue is fixed

**Comment by saveriomiroddi** [comment]
:+1:

**Comment by st0012** [comment]
@saveriomiroddi Can you take a look at this [CI failure](https://travis-ci.org/saveriomiroddi/goby-dev/builds/303642082?utm_source=email&utm_medium=notification)? I think it's because I clicked on update branch but I have no right to write your fork.
So you can rebase this on master and merge it if the CI passed.

File: vm/class.go [hunk 1]
```diff
@@ -1024,10 +1024,23 @@ func builtinClassCommonInstanceMethods() []*BuiltinMethodObject {
...
+					if ok {
+						seconds := int.value
+						time.Sleep(time.Duration(seconds) * time.Second)
+						return int
+					}
+
+					float, ok := args[0].(*FloatObject)
+
+					if ok {
+						nanoseconds := int64(float.value * 1000000000)
```
Comment (st0012) [inline vm/class.go]: Can this number be set as a constant?

Comment (saveriomiroddi) [reply vm/class.go]: Hmmm, there may be a more idiomatic way to express that. Hold on, I will check if I can improve it.

**Comment by saveriomiroddi** [comment]
There you go, I've found a more expressive way for that statement.

**Pull request merged**